src/lib.rs

```diff
@@ -90,20 +91,17 @@ pub fn run_scheduler(
     goals: &[Goal],
 ) -> FinalTasks {
     let mut calendar = Calendar::new(start_date, end_date);
-    dbg!(&calendar);
 
     calendar.add_budgets_from(goals);
 
     //generate and place simple goal activities
     let simple_goal_activities =
         activity_generator::generate_simple_goal_activities(&calendar, goals);
-    dbg!(&simple_goal_activities);
+    let simple_goal_activities = adjust_parent_activities(&simple_goal_activities, goals);
```

Comment:
Please make this a responsibility of fn `generate_simple_goal_activities`

Reply:
Sorry for late response.

---
I did the change but it causing wrong scheduling for `budget_and_goal_one_day`
```shell
running 10 tests
test e2e::basic_1 ... ok
test e2e::filler_goal ... ok
test e2e::algorithm_challenge ... ok
test e2e::budget_and_goal_one_day ... FAILED
test e2e::bug_start_missing ... ok
test e2e::not_on ... ok
test e2e::after_12 ... ok
test e2e::sleep_1 ... ok
test e2e::budget_with_subgoal ... ok
test e2e::default_budgets ... ok

```

I have a challenge here.

I did it but it causing wrong scheduling although I have the same params.

I did investigate it and found below:
- When calling `adjust_parent_activities` from inside `generate_simple_goal_activities`, it causing wrong values for `calender_overlay` as below figure:
![image](https://github.com/tijlleenders/ZinZen-scheduler/assets/5870208/de8c9c05-2cfb-4739-838f-c2ec613ece06)

- When calling `adjust_parent_activities` from inside `lib.rs` (like current case), it causing accurate values which leads to accurate scheduling as below: 
![image](https://github.com/tijlleenders/ZinZen-scheduler/assets/5870208/7f94f237-2e32-4d65-bc87-da2661d559da)

---

So, I suggest to avoid investigate this issue at the time and solve more issues to achieve business requirements. Also I'm ready to continue investigating and resolving this issue based on your request.

Reply:
Please investigate what the issue is and propose a fix.  

I don't see why activities generated from budget goals should be affected by generating/adjusting activities from simple goals.

Ignoring the problem is not a solution.

Reply:
I agree...
I will, thanks